tests/resource/test_resource.py

```diff
@@ -0,0 +1,127 @@
+# Copyright 2018-2023 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+Test base Resource class and its associated methods
+"""
+from dataclasses import FrozenInstanceError
+import pytest
+
+from pennylane.resource import Resources
+
+
+class TestResources:
```

Comment:
Are these the original tests used before decorating `Resource` with `@dataclass(frozen=True)`? Or the tests had to be modified?

Reply:
Same tests, they don't need to be modified.